pkg/machine/qemu/machine.go

```diff
@@ -327,17 +300,35 @@ func (v *MachineVM) Init(opts machine.InitOptions) (bool, error) {
 		logrus.Warn("ignoring init option to disable user-mode networking: this mode is not supported by the QEMU backend")
 	}
 
+	builder := machine.NewIgnitionBuilder(machine.DynamicIgnition{
+		Name:      opts.Username,
```

Comment:
@jakecorrenti wwyt about something where you get the builder back, and then do something like:

`builder.WithUnit(suchAndSuch)` ?  Lets sync up and brain storm around this.  I was expecting that but maybe you are on the better track too!  @mheon @Luap99 wdyt?

Reply:
Is there an advantage to that over the current struct-based approach? I'd be fine merging this as-is.

Reply:
@baude we can sync up on this today if you have the time. I'm indifferent about implementation as long as the change is made and everyone's happy with it.